state-chain/amm/src/tests.rs

```diff
@@ -41,25 +38,21 @@ fn test_basic_swaps() {
 			assert_eq!(
 				assert_ok!(pool_state.collect_and_mint_limit_order(
 					&LiquidityProvider::from([0; 32]),
-					!asset,
-					Order::Sell,
+					!order,
 					0,
 					amount
 				)),
 				(Default::default(), limit_orders::PositionInfo::new(amount))
 			);
 
-			assert_eq!(pool_state.swap(asset, Order::Sell, 0.into(), None), (0.into(), 0.into()));
-			assert_eq!(
-				pool_state.swap(asset, Order::Sell, Amount::MAX, None),
-				(amount, Amount::MAX - amount)
-			);
+			assert_eq!(pool_state.swap(order, 0.into(), None), (0.into(), 0.into()));
+			assert_eq!(pool_state.swap(order, Amount::MAX, None), (amount, Amount::MAX - amount));
 		}
 
 		{
-			let initial_sqrt_price = match asset {
-				common::Side::Zero => MAX_SQRT_PRICE - 1,
```

Comment:
did we want to remove the '-1' ?

Reply:
yes (should have been done before in previous pr, saw I hadn't before)

Reply:
I increase the vlid sqrt_prices to include the MAX_SQRT_PRICE in all cases, previously so some of internal functions where safer to use. As a result MAX_SQRT_PRICE is a valid starting price. Where it wasn't before.